pkg/api/v20240812preview/openshiftcluster_validatestatic.go

```diff
@@ -446,6 +446,29 @@ func (sv openShiftClusterStaticValidator) validateDelta(oc, current *OpenShiftCl
 		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodePropertyChangeNotAllowed, err.Target, err.Message)
 	}
 
+	if current.UsesWorkloadIdentity() {
+		currentIdentities := map[string]PlatformWorkloadIdentity{}
+		for _, i := range current.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities {
+			currentIdentities[i.OperatorName] = i
+		}
+
+		updateIdentities := map[string]PlatformWorkloadIdentity{}
+		for _, i := range oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities {
+			updateIdentities[i.OperatorName] = i
```

Comment:
I know we handle the duplicate PlatformWorkloadIdentities at CLI side, if we want to handle the duplication error in the provided list of identities, this would be the best place to handle it.
@tsatam @cadenmarchese , do you agree handling it here?

Reply:
+1 on handling this here (checking for duplicates in OperatorName). I've added it to the JIRA.

Reply:
I think this still needs handling here:-

```
_ , duplicate := updateIdentities[i.OperatorName]
if duplicate {
				return api.NewCloudError(http.StatusBadRequest, api.[SOME-CLOUD-ERROR-FOR-NO-DUPLICATE-OPERATOR-IDENTITIES], "properties.platformWorkloadIdentityProfile.platformWorkloadIdentities", "<SOME MESSAGE FOR DUPLICATE OPERATOR IDENTITIES>")
			}
```